src/hyperion/parameters/rotation.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+import os
+
+import numpy as np
+from dodal.devices.detector import DetectorParams
+from dodal.devices.detector.det_dist_to_beam_converter import (
+    DetectorDistanceToBeamXYConverter,
+)
+from dodal.devices.zebra import (
+    RotationDirection,
+)
+from pydantic import Field
+from scanspec.core import AxesPoints
+from scanspec.core import Path as ScanPath
+from scanspec.specs import Line
+
+from hyperion.external_interaction.ispyb.ispyb_dataclass import RotationIspybParams
+from hyperion.parameters.components import (
+    DiffractionExperiment,
+    OptionalGonioAngleStarts,
+    OptionalXyzStarts,
+    RotationAxis,
+    TemporaryIspybExtras,
+    WithSample,
+    WithScan,
+)
+from hyperion.parameters.constants import CONST
+from hyperion.parameters.plan_specific.rotation_scan_internal_params import (
+    RotationHyperionParameters,
+    RotationInternalParameters,
+    RotationScanParams,
+)
+
+
+class RotationScan(
+    DiffractionExperiment,
+    WithScan,
+    OptionalGonioAngleStarts,
+    OptionalXyzStarts,
+    WithSample,
+):
+    omega_start_deg: float = Field(default=0)  # type: ignore
+    rotation_axis: RotationAxis = Field(default=RotationAxis.OMEGA)
+    shutter_opening_time_s: float = Field(default=CONST.I03.SHUTTER_TIME_S)
+    rotation_angle_deg: float
+    rotation_increment_deg: float
+    rotation_direction: RotationDirection
+    transmission_frac: float
+    ispyb_extras: TemporaryIspybExtras
+
+    @property
+    def directory(self):
+        directory = str(self.visit_directory / "auto" / str(self.sample_id))
```

Comment:
Must: This certainly isn't right. From looking round on the file system it looks like some use sample name some use something else. We may have to provide the `prefix` as in http://agamemnon.diamond.ac.uk/example/collect for now then discuss with SIMS/Analysis how that is actually built.

Reply:
We could just redefine it for UDC and say it is always the sample_id but we should shout a bit louder before we do that